readthedocs/search/parsers.py

```diff
@@ -307,10 +308,42 @@ def parse(self, page):
             'domain_data': {},
         }
         """
-        raise NotImplementedError
+        try:
+            content = self._get_page_content(page)
+            if content:
+                return self._process_content(page, content)
+        except Exception as e:
+            log.info("Failed to index page.", path=page, exc_info=True)
+        return {
+            "path": page,
+            "title": "",
+            "sections": [],
+            "domain_data": {},
+        }
 
+    def _process_content(self, page, content):
+        """Parses the content into a structured dict."""
+        html = HTMLParser(content)
+        body = self._get_main_node(html)
+        title = ""
+        sections = []
+        if body:
+            title = self._get_page_title(body, html) or page
+            sections = self._get_sections(title=title, body=body)
+        else:
+            log.info(
```

Comment:
I do wonder if there is a way to surface this to users? I imagine especially with `build.commands`, users are going to want to have their content indexed, but not know how to do it. To start with we should document our feature here, and how we search for content to index, and perhaps eventually raise these errors in the build somehow in the future.